### PR TITLE
user-word-list

### PR DESCRIPTION
**Pull request by phamhuycuong**

### CODE REVIEW

File: app/controllers/words_controller.rb [hunk 1]
```diff
@@ -1,8 +1,41 @@
+require "will_paginate/array"
```
Comment (nguyenduyta) [inline app/controllers/words_controller.rb]: ? anh require de lam gi vay a?

Comment (phamhuycuong) [inline app/controllers/words_controller.rb]: de paginate array, phan alphabet dua ra array.

File: app/controllers/words_controller.rb [hunk 2]
```diff
@@ -1,8 +1,41 @@
...
   def index
-    @words = Word.paginate page: params[:page], per_page: Settings.words_paginate
+    if ["search"].include? params[:search]
+      if 0 == params[:category].to_i
+        @words = Word.all
+      else
+        @words = Category.find(params[:category].to_i).words
+      end
+
+      case params[:learn]
```
Comment (phamkykhoi) [inline app/controllers/words_controller.rb]: Xử lý phức tạp

File: app/views/words/_show.html.erb [hunk 3]
```diff
@@ -1,30 +1,11 @@
 <div class="row">
   <div class="col-md-6 col-md-offset-3">
-    Category: <strong><%= Category.find(@word.category_id).title.capitalize %> </strong>
+    <%= I18n.t "word.category" %>:&nbsp;<strong><%= Category.find(word.category_id).title.capitalize %> </strong>
```
Comment (phamkykhoi) [inline app/views/words/_show.html.erb]: use Scope

File: app/views/words/_list_words_head.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<div class="row">
+  <div class="col-md-2">
+  </div>
+  <div class="col-md-2">
+    <%= I18n.t "word.category" %>
+    <select name="category" >
+      <option value="0"><%= I18n.t "word.all" %></option>
+      <% Category.all.each do |category| %>
+        <option value="<%= category.id %>"><%= category.title %></option>
```
Comment (phamkykhoi) [inline app/views/words/_list_words_head.html.erb]: indent

File: app/views/words/_list_words_head.html.erb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  </div>
+  <div class="col-md-2">
+    <%= I18n.t "word.sort_by" %>
+    <select name="sort_by" >
+      <option value="id"><%= I18n.t "word.id" %></option>
+      <option value="alphabet"><%= I18n.t "word.alphabet" %></option>
+    </select>
+  </div>
+  <div class="col-md-1" align="center">
+    <%= I18n.t "word.passed" %><input type="radio" name="learn" value="passed"/>
```
Comment (phamkykhoi) [inline app/views/words/_list_words_head.html.erb]: Duplicate code. use array

File: app/views/words/_words.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  <div class="col-md-2">
+  </div>
+  <div class="col-md-8">
+    <table class="table table-hover w100">
+      <tr>
+        <td width="5%"><strong><%= I18n.t "word.id" %></trong></td>
+        <td><strong><%= I18n.t "word.content" %></strong></td>
+      </tr>
+      <% words.each do |word| %>
+        <p>
```
Comment (phamkykhoi) [inline app/views/words/_words.html.erb]: Remove tag p

File: app/controllers/words_controller.rb (hunk 1 above)
Comment (phamkykhoi) [inline app/controllers/words_controller.rb]: Remove

File: app/controllers/words_controller.rb [hunk 7]
```diff
@@ -1,8 +1,41 @@
+require "will_paginate/array"
+
 class WordsController < ApplicationController
   before_action :set_word, except: [:new, :index, :create]
 
   def index
-    @words = Word.paginate page: params[:page], per_page: Settings.words_paginate
+    if ["search"].include? params[:search]
+      if 0 == params[:category].to_i
```
Comment (khanhkhanh) [inline app/controllers/words_controller.rb]: Try using @words = params[:category_id] == 0 ? Word.all : Category.find(category_id).words
(maybe you have to add () for syntax)

File: app/views/admin/_word_list.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<h1><%= I18n.t :listing_words %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <% words.each do |word| %>
```
Comment (khanhkhanh) [inline app/views/admin/_word_list.html.erb]: Move this line into <table> tags

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -1,10 +1,11 @@
...
+    <%= link_to I18n.t(:menu_project_name), "#", id: "logo" %>
     <nav>
       <ul class="nav navbar-nav navbar-right">
-        <li><%= link_to I18n.t(:menu_home),   "#" %></li>
-        <li><%= link_to I18n.t(:menu_help),   "#" %></li>
+        <li><%= link_to I18n.t(:menu_home), root_path %></li>
+        <li><%= link_to I18n.t(:menu_help), "#" %></li>
+        <li><%= link_to I18n.t(:menu_categories), categories_path %></li>
         <% if logged_in? %>
           <% if current_user.admin %>
```
Comment (khanhkhanh) [inline app/views/layouts/_header.html.erb]: current_user.admin?

File: app/views/words/_show.html.erb (hunk 3 above)
Comment (khanhkhanh) [inline app/views/words/_show.html.erb]: word.category.title.capitalize

File: app/views/admin/_word.html.erb [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= I18n.t "word.category" %>:&nbsp;<strong><%= Category.find(word.category_id).title.capitalize %> </strong>
```
Comment (khanhkhanh) [inline app/views/admin/_word.html.erb]: Category.find(word.category_id).title.capitalize => word.category

File: app/controllers/words_controller.rb [hunk 11]
```diff
@@ -2,7 +2,23 @@ class WordsController < ApplicationController
   before_action :set_word, except: [:new, :index, :create]
 
   def index
-    @words = Word.paginate page: params[:page], per_page: Settings.words_paginate
+    if ["search"].include? params[:search]
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: `params[:search].present?`

File: app/controllers/words_controller.rb [hunk 12]
```diff
@@ -2,7 +2,14 @@ class WordsController < ApplicationController
   before_action :set_word, except: [:new, :index, :create]
 
   def index
-    @words = Word.paginate page: params[:page], per_page: Settings.words_paginate
+    if params[:search].present?
+      @words = Word.filter_category(params[:category_id].to_i).send(params[:learn].downcase, current_user)
```
Comment (khanhkhanh) [inline app/controllers/words_controller.rb]: Remove .to_i
And try changing send(params[:learn].downcase, current_user) to send params[:learn].downcase, current_user

File: app/controllers/words_controller.rb [hunk 13]
```diff
@@ -2,7 +2,14 @@ class WordsController < ApplicationController
...
+      @words = @words.sort_by{|word| word.content.downcase} if "alphabet" == params[:sort_by]
```
Comment (khanhkhanh) [inline app/controllers/words_controller.rb]: scope

File: app/views/words/_list_words_head.html.erb [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        <input type="radio" name="learn" value="<%= radio_button %>"/>
+      <% end %>
+    </div>
+  <% end %>
+</div>
+<div class="row">
+  <div class="col-md-2">
+    </div>
+    <div class="col-md-4">
+        <input type="text" name="word_search" class="form-control"/>
```
Comment (phamkykhoi) [inline app/views/words/_list_words_head.html.erb]: Indent

File: app/views/words/_list_words_head.html.erb [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        <input type="radio" name="learn" value="get_all" checked="on"/>
+      <% else %>
+        <input type="radio" name="learn" value="<%= radio_button %>"/>
+      <% end %>
+    </div>
+  <% end %>
+</div>
+<div class="row">
+  <div class="col-md-2">
+    </div>
```
Comment (phamkykhoi) [inline app/views/words/_list_words_head.html.erb]: Move up

File: app/views/words/_list_words_head.html.erb [hunk 16]
```diff
@@ -0,0 +1,42 @@
+<div class="row">
+  <div class="col-md-2">
+  </div>
+  <div class="col-md-2">
+    <%= I18n.t "word.category" %>
+    <select name="category_id" >
+      <option value="0"><%= I18n.t "word.all" %></option>
+      <% Category.all.each do |category| %>
```
Comment (khanhkhanh) [inline app/views/words/_list_words_head.html.erb]: Move it into controller

File: app/controllers/words_controller.rb [hunk 17]
```diff
@@ -2,7 +2,15 @@ class WordsController < ApplicationController
   before_action :set_word, except: [:new, :index, :create]
 
   def index
-    @words = Word.paginate page: params[:page], per_page: Settings.words_paginate
+    if params[:search].present?
+      @words = Word.filter_category(params[:category_id].to_i).send params[:learn].downcase, current_user
```
Comment (phamkykhoi) [inline app/controllers/words_controller.rb]: If params[:category_id] == empty. What happen ?

File: app/views/words/_list_words_head.html.erb [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+      <option value="id"><%= I18n.t "word.id" %></option>
+      <option value="alphabet"><%= I18n.t "word.alphabet" %></option>
+    </select>
+  </div>
+  <% radio_button_array = [I18n.t("word.passed"), I18n.t("word.learned"), I18n.t("word.unlearned"), I18n.t("word.all")] %>
+  <% radio_button_array.each do |radio_button| %>
+    <div class="col-md-1" align="center">
+      <%= radio_button %>
+      <% if "All" == radio_button %>
+        <input type="radio" name="learn" value="get_all" checked="on"/>
```
Comment (phamkykhoi) [inline app/views/words/_list_words_head.html.erb]: Keep value when submit form filter

**Comment by phamkykhoi** [comment]
OK

**Comment by dieunb** [comment]
OK

**Pull request merged**